lib/logging/third_party_transaction.rb

```diff
@@ -61,14 +70,35 @@ def log_3pi_complete(method_name, additional_logs = {}, args = [])
           action: 'Complete interaction with 3rd party API',
           wrapped_method: "#{self.class}##{method_name}",
           upload_duration: (now - @start_time).to_f,
-          end_time: now,
+          end_time: now.to_s,
           passed_args: args.to_s
-        }.merge(additional_logs)
+        }.merge(additional_class_logs).merge(parse_instance_logs(additional_instance_logs))
 
         Rails.logger.info(log)
       rescue => e
         Rails.logger.error(e)
       end
+
+      def parse_instance_logs(additional_instance_logs)
+        # call method chains on instance and return a hash of values, e.g.
+        # HAPPY PATH (where my_instance.user_account.id == s0meUs3r-Id-V@Lu3')
+        # { user_uuid: [:user_account, :id] }
+        # will be translated to
+        # { user_uuid: 's0meUs3r-Id-V@Lu3' }
+        #
+        # or
+        #
+        # SAD PATH (safe)
+        # { user_uuid: [:user_account, :id, :non_working_method, :something_else_that_breaks] }
+        # will be translated to
+        # { user_uuid: nil }
+        {}.tap do |obj|
+          additional_instance_logs.each do |key, method_chain|
+            # using :try ensures we can fail quietly
+            obj[key] = method_chain.inject(self, :try)
```

Comment:
What will happen if/when we fail quietly here?

Reply:
we will just log a nil value.  This is a feature, allowing people to pass method chains that may be unstable due to context, such as inheritance.  Also, just a safe gaurd so logging doesn't break important actions

Reply:
Note that logging a nil value for one method chain will *not* affect the other log values